✨ feat: center spider element using CSS positioning

de-vasta opened the pull request:
[DEMO LINK](https://de-vasta.github.io/js_center_spider_DOM/)

de-vasta committed:
✨ feat: center spider element using CSS positioning

- Update demo link with actual GitHub username
- Implement spider centering with `inset: 0` and `margin: auto`
- Fix minor formatting issues in README

mateacademy-ai-mentor commented on src/scripts/main.js:
```diff
@@ -1,3 +1,6 @@
+const spider = document.querySelector('.wall > .spider');
+
+spider.style.inset = '0';
+spider.style.margin = 'auto';
```
This approach centers the spider using a CSS technique. However, the task requires you to manually calculate the position. You should get the dimensions of the `.wall` and `.spider` elements and use them to calculate the correct `top` and `left` CSS properties.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Great job getting the spider visually centered on the wall! The outcome looks perfect. However, I am requesting changes because the implementation doesn't follow the core requirement of the task. The instructions specify that you must *calculate* the spider's position using the dimensions of the parent (`.wall`) and the spider itself. Your current solution uses a CSS-based centering technique (`inset: 0` and `margin: auto`) applied via JavaScript, which bypasses this essential calculation step. Please adjust your code to get the dimensions of the elements (e.g., using `offsetWidth` and `offsetHeight`) and then compute the `top` and `left` properties mathematically.

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

de-vasta committed:
♻️ refactor(spider): improve centering calculation using scroll dimensions

- Replace CSS-based centering with JavaScript positioning
- Calculate center using `scrollHeight` and `scrollWidth` properties

mateacademy-ai-mentor commented on src/scripts/main.js:
```diff
@@ -1,3 +1,8 @@
+
+spider.style.top = (wall.scrollHeight - spider.scrollHeight) / 2 + 'px';
+
+spider.style.left = (wall.scrollWidth - spider.scrollWidth) / 2 + 'px';
```
This is a great step forward in calculating the position mathematically! However, using `scrollHeight` and `scrollWidth` can lead to incorrect centering if the element's content overflows its visible area. 

For more reliable layout calculations, it's better to use properties that reflect the element's rendered size, such as `offsetHeight` and `offsetWidth` (which include padding and borders) or `clientHeight` and `clientWidth` (which include padding).

de-vasta replied:
@mateacademy-ai-mentor `scrollWidth` and `scrollHeight` the only ones centering inaccurately

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Excellent work on implementing the mathematical calculation for the spider's position! This is a huge step forward. I am requesting one small but important change to make the solution more robust. Your calculation currently uses `scrollHeight` and `scrollWidth`, which can be unreliable for positioning as they include the size of non-visible, overflowing content. For more accurate and stable centering, please update your code to use properties that reflect the element's rendered size, such as `offsetHeight`/`offsetWidth` or `clientHeight`/`clientWidth`. After this small adjustment, your solution will be perfect!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

de-vasta committed:
♻️ refactor(spider): use clientHeight/clientWidth for centering

- Replace `scrollHeight`/`scrollWidth` with `clientHeight`/`clientWidth`
- Ensures proper centering within visible viewport area

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Excellent work on this revision! Your switch to using `clientHeight` and `clientWidth` for the calculations is the perfect solution and fully addresses the previous feedback. The code is now robust, clean, and correctly centers the spider within the field. Your solution now meets all the requirements of the task. Great job!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)
